game_engine.hpp

```diff
@@ -14,11 +15,16 @@ template <std::size_t RowCount, std::size_t ColumnCount>
 class game_engine
 {
 public:
-    CONSTEXPR game_engine(const board<RowCount, ColumnCount>& board, int score, int moves, long long epoch_ms) : 
-        board_(board),
+    CONSTEXPR game_engine(const std::tuple< std::array<std::array<candy, ColumnCount>, RowCount>, int, int, bool, int, int > board_extended, int score, int moves, long long epoch_ms) :
```

Comment:
Although I really like tuples, this might be worth creating a proper `struct`.

Reply:
Agree! I also thought about using struct, tuple won in choice between rationally good code and cool over-engineered piece. I'll implement struct, later, now at 2:30 A.M. risk for bugs is too high.

Reply:
I made it in time before PR merge! You can now merge version with `struct` instead for `tuple` as you suggested. I can think better at 17:30 than at 2:30 :-)

Reply:
hehe good job here @aexg !